crates/core/tedge_api/src/workflow/supervisor.rs

```diff
@@ -76,7 +76,25 @@ impl WorkflowSupervisor {
     }
 
     /// Update on start the set of pending commands
-    pub fn load_pending_commands(&mut self, commands: CommandBoard) -> Vec<GenericCommandState> {
+    pub fn load_pending_commands(
+        &mut self,
+        mut commands: CommandBoard,
+    ) -> Vec<GenericCommandState> {
+        // If the resumed commands have been triggered by an agent without workflow version management
+        // then these commands are assigned the current version for the operation.
+        // These currents versions have also to be marked as in use.
+        for (_, ref mut command) in commands.iter_mut() {
+            if command.workflow_version().is_none() {
+                if let Some(versions) = command
+                    .operation()
+                    .and_then(|operation| self.workflows.get_mut(&operation.as_str().into()))
+                {
+                    if let Some(current_version) = versions.use_current_version() {
+                        *command = command.clone().set_workflow_version(current_version);
```

Comment:
Not so nice, but okayish as this happens only on very specific and rare cases (restart after a self update).

It would be nicer to change `set_workflow_version` signature:

```suggestion
                        command.set_workflow_version(current_version);
```